sdk/Javascript/src/Platforms/Platform.FxOS.js

```diff
@@ -0,0 +1,277 @@
+                    complete.apply(null, Array.prototype.slice.call(arguments, 1));
+                } else {
+                    error(err);
+                }
+            };
+            Array.prototype.push.call(args, callback);
+
+            try {
+                // Invoke the async method which will in turn invoke our callback
+                // which will in turn invoke the promise's handlers
+                func.apply(that, args);
+            } catch (ex) {
+                // Thread any immediate errors like parameter validation
+                // through the the callback
+                callback(_.createError(ex));
+            }
+        });
+    };
+};
+
+exports.addToMobileServicesClientNamespace = function (declarations) {
+    /// <summary>
+    /// Define a collection of declarations in the Mobile Services Client namespace.
+    /// </summary>
+    /// <param name="declarations" type="Object">
+    /// Object consisting of names and values to define in the namespace.
+    /// </param>
+
+    // First ensure our 'WindowsAzure' namespace exists
+    var namespaceObject = global.WindowsAzure = global.WindowsAzure || {};
+    
+    // Now add each of the declarations to the namespace
+    for (var key in declarations) {
+        if (declarations.hasOwnProperty(key)) {
+            namespaceObject[key] = declarations[key];
+        }
+    }
+};
+
+exports.readSetting = function readSetting(name) {
+    /// <summary>
+    /// Read a setting from a global configuration store.
+    /// </summary>
+    /// <param name="name" type="String">
+    /// Name of the setting to read.
+    /// </param>
+    /// <returns type="String" mayBeNull="true">
+    /// The value of the setting or null if not set.
+    /// </returns>
+
+    return inMemorySettingStore[name];
+};
+
+exports.writeSetting = function writeSetting(name, value) {
+    /// <summary>
+    /// Write a setting to a global configuration store.
+    /// </summary>
+    /// <param name="name" type="String">
+    /// Name of the setting to write.
+    /// </param>
+    /// <param name="value" type="String" mayBeNull="true">
+    /// The value of the setting.
+    /// </returns>
+
+    inMemorySettingStore[name] = value;
+};
+
+exports.webRequest = function (request, callback) {
+    /// <summary>
+    /// Make a web request.
+    /// </summary>
+    /// <param name="request" type="Object">
+    /// Object describing the request (in the WinJS.xhr format).
+    /// </param>
+    /// <param name="callback" type="Function">
+    /// The callback to execute when the request completes.
+    /// </param>
+
+    return getBestTransport().performRequest(request, callback);
+};
+
+exports.getUserAgent = function () {
+    // Browsers don't allow you to set a custom user-agent in ajax requests. Trying to do so
+    // will cause an exception. So we don't.
+    return null;
+};
+
+exports.getOperatingSystemInfo = function () {
+    return {
+        name: "--",
```

Comment:
Ideally we can populate these on FireFox OS?  See the WinJS version for how we do it for Windows OS apps.